code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -264,3 +264,278 @@
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0 //Don't bring a sword to a gunfight
+	return ..()
+
+/obj/item/cursed_katana/proc/check_input(mob/living/target, mob/user)
+	for(var/combo in combo_list)
+		var/list/combo_specifics = combo_list[combo]
+		if(compare_list(input_list,combo_specifics[COMBO_STEPS]))
+			INVOKE_ASYNC(src, combo_specifics[COMBO_PROC], target, user)
+			return TRUE
+	return FALSE
+
+/obj/item/cursed_katana/proc/reset_inputs(mob/user, deltimer)
+	input_list.Cut()
+	if(user)
+		to_chat(user, "<span class='notice'>You return to neutral stance</span>")
+	if(deltimer && timerid)
+		deltimer(timerid)
+
+/obj/item/cursed_katana/proc/strike(mob/living/target, mob/user)
+	user.visible_message("<span class='warning'>[user] strikes [target] with [src]'s hilt!</span>",
+		"<span class='notice'>You hilt strike [target]!</span>")
+	to_chat(target, "<span class='userdanger'>You've been struck by [user]!</span>")
+	playsound(src, 'sound/weapons/genhit3.ogg', 50, TRUE)
+	RegisterSignal(target, COMSIG_MOVABLE_IMPACT, .proc/strike_throw_impact)
+	var/atom/throw_target = get_edge_target_turf(target, user.dir)
+	target.throw_at(throw_target, 5, 3, user, FALSE)
+	target.apply_damage(17, BRUTE, "chest")
+	to_chat(target, "<span class='userdanger'>You've been struck by [user]!</span>")
+	user.do_attack_animation(target, ATTACK_EFFECT_PUNCH)
+
+/obj/item/cursed_katana/proc/strike_throw_impact(mob/living/source, atom/hit_atom, datum/thrownthing/thrownthing)
+	SIGNAL_HANDLER
+
+	UnregisterSignal(source, COMSIG_MOVABLE_IMPACT)
+	source.apply_damage(5, BRUTE, "chest")
+	if(ishostile(source))
+		var/mob/living/simple_animal/hostile/target = source
+		target.ranged_cooldown += 2 SECONDS
+	else if(iscarbon(source))
+		var/mob/living/carbon/target = source
+		target.AdjustConfused(8 SECONDS)
+	return NONE
+
+/obj/item/cursed_katana/proc/slice(mob/living/target, mob/user)
+	user.visible_message("<span class='warning'>[user] does a wide slice!</span>",
+		"<span class='notice'>You do a wide slice!</span>")
+	playsound(src, 'sound/weapons/bladeslice.ogg', 50, TRUE)
+	var/turf/user_turf = get_turf(user)
+	var/dir_to_target = get_dir(user_turf, get_turf(target))
+	var/static/list/cursed_katana_slice_angles = list(0, -45, 45, -90, 90) //so that the animation animates towards the target clicked and not towards a side target
+	for(var/iteration in cursed_katana_slice_angles)
+		var/turf/turf = get_step(user_turf, turn(dir_to_target, iteration))
+		user.do_attack_animation(turf, ATTACK_EFFECT_CLAW)
+		for(var/mob/living/additional_target in turf)
+			if(user.Adjacent(additional_target) && additional_target.density)
+				additional_target.apply_damage(15, BRUTE, "chest", TRUE)
+				to_chat(additional_target, "<span class='userdanger'>You've been sliced by [user]!</span>")
+	target.apply_damage(5, BRUTE, "chest", TRUE)
```

Comment:
I would prefer we not name variables `turf`

Reply:
can be renamed

Reply:
renamed